### PR TITLE
Fix buffer tests memory leaks

### PR DESCRIPTION
**Pull request by SebastianLuzynski**
This change fixes buffer tests, broken by PR #1082. This pull request is similar to #1160.

### CODE REVIEW

**Comment by gwawiork** [comment]
This need to be fixed I see errors in test `test_buffers buffer_read_int`

File: test_conformance/buffers/test_buffer_read.cpp [hunk 1]
```diff
@@ -678,33 +677,41 @@ int test_buffer_read( cl_device_id deviceID, cl_context context, cl_command_queu
...
+            err = clSetKernelArg(kernel[i], 0, sizeof(cl_mem), (void *)&buffer);
             if ( err != CL_SUCCESS ){
                 print_error( err, "clSetKernelArg failed" );
                 align_free( outptr[i] );
                 align_free( inptr[i] );
                 return -1;
             }
 
-            err = clEnqueueNDRangeKernel( queue, kernel[i], 1, NULL, global_work_size, NULL, 0, NULL, NULL );
+            err = clEnqueueReadBuffer(queue, buffer, CL_TRUE, 0,
```
Comment (gwawiork) [inline test_conformance/buffers/test_buffer_read.cpp:702]: This change looks strange

Comment (SebastianLuzynski) [reply test_conformance/buffers/test_buffer_read.cpp:702]: Clang-format threw this. I had to change it so that checks will pass.

**Comment by SebastianLuzynski** [comment]
> This need to be fixed I see errors in test `test_buffers buffer_read_int`

I've run it several times and everything passes. Do you have the newest version of the repo? Pull request #1160 has been merged yesterday and it provides fixes for some tests.

File: test_conformance/buffers/test_buffer_read.cpp (hunk 1 above)
Comment (SebastianLuzynski) [reply test_conformance/buffers/test_buffer_read.cpp]: My bad, I didn't notice I've changed a function here.

**Review by EwanC — APPROVED** [review]

**Review by gwawiork — APPROVED** [review]
LGTM

**Pull request merged**